python/modules/HNLReconstruction.py

```diff
@@ -0,0 +1,149 @@
+            
+            
+        if len(otherJets)==0:
+            self.out.fillBranch(self.outputName+"_nTrueMissed", 0)
+        else:
+            missed = sum(map(lambda x: x.isLLP_Q+x.isLLP_QE+x.isLLP_QMU+x.isLLP_QTAU,otherJets))
+            self.out.fillBranch(self.outputName+"_nTrueMissed", missed)
+
+    def analyze(self, event):
+
+        lepton1 = self.lepton1Object(event)
+        lepton2 = None if self.lepton2Object==None else self.lepton2Object(event)
+          
+        jets = self.jetCollection(event)
+
+        if lepton2==None:
+            if len(jets) > 0:
+                #take jet opposite of l1 in transverse plane
+                sortedJets = sorted(jets, key=lambda jet: math.fabs(deltaPhi(jet, lepton1)), reverse=True)
+                hnlJet = sortedJets[0]
+                self.out.fillBranch(self.outputName+"_m_l1j", (lepton1.p4()+hnlJet.p4()).M())
+                self.out.fillBranch(self.outputName+"_deltaPt_l1j", (lepton1.p4()-hnlJet.p4()).Pt())
+                self.out.fillBranch(self.outputName+"_deltaPhi_l1j", math.fabs(deltaPhi(lepton1, hnlJet)))
+                self.out.fillBranch(self.outputName+"_deltaR_l1j", deltaR(lepton1, hnlJet))
+                self.out.fillBranch(self.outputName+"_deltaEta_l1j", math.fabs(lepton1.eta-hnlJet.eta))
+                setattr(event, "hnlJets_"+self.outputName, [hnlJet])
+                
+                if self.globalOptions["isSignal"]:
+                    self.fillTruthInfo(hnlJet,sortedJets[1:])
+                
+            else:
+                self.out.fillBranch(self.outputName+"_m_lj", 0)
+                self.out.fillBranch(self.outputName+"_deltaPt_lj", 0)
+                self.out.fillBranch(self.outputName+"_deltaPhi_l1j", 0)
+                self.out.fillBranch(self.outputName+"_deltaR_l1j", 0)
+                self.out.fillBranch(self.outputName+"_deltaEta_l1j", 0)
+                setattr(event, "hnlJets_"+self.outputName, [])
+                
+                if self.globalOptions["isSignal"]:
+                    self.fillTruthInfo(None,[])
+        else:
+            if len(jets) > 0:
+                #take jet closest to l2
+                sortedJets = sorted(jets, key=lambda jet: deltaR(jet, lepton2), reverse=False)
+                hnlJet = sortedJets[0]
+                self.out.fillBranch(self.outputName+"_m_llj", (lepton1.p4()+lepton2.p4()+hnlJet.p4()).M())
+                self.out.fillBranch(self.outputName+"_deltaPt_llj", (lepton1.p4()-lepton2.p4()-hnlJet.p4()).Pt())
+                self.out.fillBranch(self.outputName+"_deltaPhi_l1j", math.fabs(deltaPhi(lepton1,hnlJet)))
+                self.out.fillBranch(self.outputName+"_deltaEta_l1j", math.fabs(lepton1.eta-hnlJet.eta))
+                self.out.fillBranch(self.outputName+"_deltaR_l2j", deltaR(lepton2,hnlJet))
+                setattr(event, "hnlJets_"+self.outputName, [hnlJet])
+                
+                isMerged = 1 if deltaR(hnlJet,lepton2)<0.4 else 0
```

Comment:
i would even prefer to remove ``isMerged``` then since this can be indeed defined using cuts on ``` self.outputName+"_deltaR_l2j"``` during plotting/histogramming